### ISSUE TITLE
Unexpected error with assessor-pairs

### ISSUE THREAD

**Issue by wavded** [eslint/eslint#3262]
Running `eslint@1.0.0`:

With the following config:

```json
{
  "env": {
    "node": true,
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true
  },
  "rules": {
    "accessor-pairs": 2
  }
}
```

Against the following file:

```js
let obj = { set: function () {} };
module.exports = obj;
```

I get the following error:

```
  1:10  error  Getter is not present  accessor-pairs

✖ 1 problem (1 error, 0 warnings)
```

Though I'm not using a setter here and would expect no error.

**Comment by mysticatea** [comment]
Oh... Thank you for the report.

Yes, the rule should check whether or not the object literal is inside of `Object.defineProperty`/`Object.defineProperties`/`Object.create`.

**Comment by nzakas** [comment]
It's the difference between `{ get foo(){} }` and `{ get: function () {}`, so it shouldn't matter where the object literal is being used.

**Comment by michaelficarra** [comment]
No, it does matter because the following pattern is detected by this rule:

```js
Object.defineProperty(o, p, { get: function(){ } });
```

**Comment by mysticatea** [comment]
I will work on this.

**Comment by nzakas** [comment]
@michaelficarra ah right, I completely forgot about that.
